packages/ui/src/components/Select/select.tsx

```diff
@@ -9,28 +10,54 @@ import type { TypographyTextProps } from '@open-condo/ui/src'
 import { sendAnalyticsChangeEvent, extractChildrenContent } from '@open-condo/ui/src/components/_utils/analytics'
 
 import { SELECT_TYPES } from './constants'
+import { useItems } from './hooks/useItems'
+
+import { Either } from '../_utils/types'
 
 import type { SelectProps as DefaultSelectProps } from 'antd'
 
-const SELECT_CLASS_PREFIX = 'condo-select'
-const Option: typeof DefaultSelect.Option = DefaultSelect.Option
-const OptGroup: typeof DefaultSelect.OptGroup = DefaultSelect.OptGroup
+export const SELECT_CLASS_PREFIX = 'condo-select'
+export const Option: typeof DefaultSelect.Option = DefaultSelect.Option
+export const OptGroup: typeof DefaultSelect.OptGroup = DefaultSelect.OptGroup
 
-export type OptionType = React.PropsWithChildren<{
-    label: React.ReactNode
+export type OptionType = {
+    label: string
     value?: string | number | null
     disabled?: boolean
     textType?: TypographyTextProps['type']
     title?: string
+    key?: React.Key
+    hidden?: boolean
+}
+
+export type OptionsGroupType = {
+    label: string
+    options: Array<OptionType>
+    key?: React.Key
+}
+
+export type OptionsItem = Either<OptionType, OptionsGroupType> | null
+
+type OnChangeType = (
+    value: OptionType['value'] | Array<OptionType['value']>,
+    option: React.PropsWithChildren<OptionType> | Array<React.PropsWithChildren<OptionType>>
+) => void
+
+/**
+ * NOTE: Ant have bad interface for Select
+ * It may work with multiple values, but it is not specified in the interface.
+ */
+type CustomSelectProps<ValueType = SelectValueTypeBase> = Either<{
+    value?: ValueType | null
```

Comment:
Why do you use `null`? Is there any logic to this?

Because `prop?: MyType` is equal to `prop: MyType | undefined`...
And yours are equal to `value: ValueType | null | undefined`. Seems like null is redundant in this scenario, isn't it?

Reply:
But we should be able to specify a value equal to `null`

For example, if we want to add an empty option

Reply:
Missed that part) Thank you for clarifying. 

But I guess in multiple select it's always is array, isn't it?
`Nothing selected = []`

Reply:
Fixed it for multiple select